oioioi/oi/management/commands/import_schools.py

```diff
@@ -1,122 +1,533 @@
+        self.log(
+            f'INFO: Read {len(curr_rspo_schools)} records from the currently imported RSPO database.'
+        )
 
-                row['postal_code'] = ''.join(row['postal_code'].split())
+        if not DRY_RUN:
+            self.save_rspo_backup(curr_rspo_schools)
 
-                for hypen in (' - ', u'\u2010'):
-                    row['city'] = row['city'].replace(hypen, '-')
-                row['city'] = row['city'].title()
+        curr_rspo_schools = [
+            translate_rspo_school(school).__dict__ for school in curr_rspo_schools
+        ]  # List of school dicts
+        last_rspo_schools = [
+            translate_rspo_school(school).__dict__ for school in last_rspo_schools
+        ]  # List of school dicts
 
-                row['province'] = row['province'].lower()
+        if first_import:
+            set_all_schools_inactive()
 
-                row['phone'] = row['phone'].split(',')[0]
-                row['phone'] = row['phone'].split(';')[0]
-                for c in ['tel.', 'fax.', '(', ')', '-', ' ']:
-                    row['phone'] = row['phone'].replace(c, '')
-                row['phone'] = row['phone'].lstrip('0')
+        for db_school in School.objects.filter(is_active=True):
+            db_school_dict = db_school.__dict__
+            curr_rspo_school = find_school(db_school_dict, curr_rspo_schools)
+            last_rspo_school = find_school(db_school_dict, last_rspo_schools)
 
-                row['email'] = row['email'].split(',')[0]
-                row['email'] = row['email'].split(';')[0]
+            curr_rspo_school_exists = curr_rspo_school is not None
+            last_rspo_school_exists = last_rspo_school is not None
 
-                school, created = School.objects.get_or_create(
-                    name=row['name'], postal_code=row['postal_code']
+            if curr_rspo_school_exists and last_rspo_school_exists:
+                db_and_curr_the_same = are_the_same(db_school_dict, curr_rspo_school)
+                db_and_last_the_same = are_the_same(db_school_dict, last_rspo_school)
+                curr_and_last_the_same = are_the_same(
+                    curr_rspo_school, last_rspo_school
                 )
-                if created:
-                    created_count += 1
 
-                for column in COLUMNS:
-                    setattr(school, column, row[column])
+                # If the MoE has made changes that we have not, then accept MoE changes.
+                if not db_and_curr_the_same and db_and_last_the_same:
+                    self.log(
+                        f'Note: The school [rspo: {db_school.rspo}] has been changed in the RSPO database. The changes have been applied.',
+                        school_before=db_school,
+                        school_after=translate_dict_school(curr_rspo_school),
+                    )
+                    update_school(curr_rspo_school, db_school)
+                    rows_affected += 1
```

Comment:
Changes from MoE may introduce new duplicates.
```suggestion
                    try:
                        with transaction.atomic():
                            update_school(curr_rspo_school, db_school)
                        rows_affected += 1
                    except utils.IntegrityError:
                        self.log(
                            f'Warning: Django found a duplicate school. Please manually check the school {db_school.rspo} data in SIO database, '
                            'current RSPO and the last backup. Something suspicious may be going on.'
                        )
```